src/main/java/org/folio/edge/sip2/repositories/ConfigurationRepository.java

```diff
@@ -51,102 +59,135 @@ public ConfigurationRepository(IResourceProvider<IRequestData> resourceProvider,
   }
 
   /**
    * Method that retrieves the configuration from a resource provider.
    *
    * @param sessionData sessionData containing tenant ID to retrieve desired tenant configuration.
-   * @param configCode config code to identify the configuration
-   * @param configName name of the configuration
-   * @param module module that the configuration was created for
-   * @return JSON object containing tenant configuration
+   * @param configParameters different set of parameters to get values from config store
+   * @return A Map of keys and JSON config objects
    */
-  public Future<JsonObject> retrieveConfiguration(SessionData sessionData,
-                                                  String module,
-                                                  String configName,
-                                                  String configCode) {
+  public Future<LinkedHashMap<String, JsonObject>> retrieveConfigurations(SessionData sessionData,
+                                           List<LinkedHashMap<String, String>> configParameters) {
     final Map<String, String> headers = new HashMap<>();
     headers.put("accept", "application/json");
 
     ConfigurationRequestData requestData = new ConfigurationRequestData(null,
-        headers, sessionData, module, configName, configCode);
+        headers, sessionData, configParameters);
 
     Future<IResource> future = resourceProvider.retrieveResource(requestData);
     return future.compose(resource -> {
       final JsonObject scConfiguration = resource.getResource();
+      JsonArray configs = scConfiguration.getJsonArray("configs");
+      int totalConfigs = configs.size();
+      if (totalConfigs >= configParameters.size()) {
+
+        LinkedHashMap<String, JsonObject> resultJsonConfigs = new LinkedHashMap<>();
+
+        for (int i = 0; i < totalConfigs; i++) {
+          JsonObject config = configs.getJsonObject(i);
+          String module = config.getString(KEY_CONFIG_MODULE);
+          String configName = config.getString(KEY_CONFIG_NAME);
+          String code = config.getString(KEY_CONFIG_CODE);
 
-      if (scConfiguration != null) {
-        JsonArray configs = scConfiguration.getJsonArray("configs");
-        if (configs.size() > 0) {
-          JsonObject firstConfig = configs.getJsonObject(0);
+          String configKey = String.format(CONFIGURATION_TEMPLATE, module, configName, code);
 
-          String acsConfigurationString = firstConfig.getString("value");
-          if (acsConfigurationString != null && !acsConfigurationString.isEmpty()) {
-            JsonObject acsConfiguration = new JsonObject(acsConfigurationString);
-            return Future.succeededFuture(acsConfiguration);
+          String configurationString = config.getString("value");
+          if (!Utils.isStringNullOrEmpty(configurationString)) {
+            JsonObject jsonConfiguration = new JsonObject(configurationString);
+            resultJsonConfigs.put(configKey,jsonConfiguration);
```

Comment:
I have a bone to pick about a build fails because of checkstyle found missing a space here or an extra space there :) But that is another totally different discussion for another day.